typescript/sdk/src/ism/CosmosNativeIsmModule.ts

```diff
@@ -134,27 +172,157 @@ export class CosmosNativeIsmModule extends HyperlaneModule<
 
     switch (ismType) {
       case IsmType.MERKLE_ROOT_MULTISIG: {
-        const { response: merkleRootResponse } =
-          await this.signer.createMerkleRootMultisigIsm({
-            validators: config.validators,
-            threshold: config.threshold,
-          });
-        return merkleRootResponse.id;
+        return this.deployMerkleRootMultisigIsm(config);
       }
       case IsmType.MESSAGE_ID_MULTISIG: {
-        const { response: messageIdResponse } =
-          await this.signer.createMessageIdMultisigIsm({
-            validators: config.validators,
-            threshold: config.threshold,
-          });
-        return messageIdResponse.id;
+        return this.deployMessageIdMultisigIsm(config);
+      }
+      case IsmType.ROUTING: {
+        return this.deployRoutingIsm(config);
       }
       case IsmType.TEST_ISM: {
-        const { response: noopResponse } = await this.signer.createNoopIsm({});
-        return noopResponse.id;
+        return this.deployNoopIsm();
       }
       default:
-        throw new Error(`ISM type ${ismType} is not supported on Cosmos`);
+        throw new Error(
+          `ISM type ${ismType} is not supported on Cosmos Native`,
+        );
     }
   }
+
+  protected async deployMerkleRootMultisigIsm(
+    config: MultisigIsmConfig,
+  ): Promise<Address> {
+    const { response } = await this.signer.createMerkleRootMultisigIsm({
+      validators: config.validators,
+      threshold: config.threshold,
+    });
+    return response.id;
+  }
+
+  protected async deployMessageIdMultisigIsm(
+    config: MultisigIsmConfig,
+  ): Promise<Address> {
+    const { response } = await this.signer.createMessageIdMultisigIsm({
+      validators: config.validators,
+      threshold: config.threshold,
+    });
+    return response.id;
+  }
```

Comment:
shouldn't we check that the threshold makes sense compared to the number of validators? like we don't have threshold set to 10 but only have 1 validator

Reply:
the cosmos-sdk module would throw if you would send a transaction with threshold being greater than the number of validators, added an assert check anyway